doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -766,58 +726,81 @@ the conversion in hypotheses :n:`{+ @ident}`.
          pattern (0 = 0).
          fold not.
 
-   .. tacv:: fold {+ @term}
+.. tacn:: pattern {+, @pattern_occs } {? @occurrences }
 
-      Equivalent to :n:`fold @term ; ... ; fold @term`.
+   .. todo PR: how do multiple pattern_occs work with @occurrences??
 
-.. tacn:: pattern @term
-   :name: pattern
+   Performs :term:`eta-expansion`.  If :n:`@occurrences` is given,
```

Comment:
```suggestion
   Performs :term:`beta-expansion`.  If :n:`@occurrences` is given,
```
eta-expansion is only applicable to terms whose type is a function, i.e., turning `f` into `fun x => f x`.  `pattern` results in a function applied to concrete arguments, not a lambda whose body is an application to the arguments to the function.

Reply:
I was going off of @Zimmi48's comment here and above: "E.g. pattern actually performs eta-expansion. And fold performs delta-expansion." at https://github.com/coq/coq/pull/13707#discussion_r568058697

Is this a bit like `fold` in that the set of one_terms are first beta-reduced?

I put it back to beta-expansion and added a few words about "in the selected hypotheses and/or goal" but I think what I did is still not right.

Reply:
I think @Zimmi48 was wrong about eta-expansion because `pattern` does not leave functions in eta-long form and eta-contraction/eta-reduction would not undo `patterm`.  

It is true that `fold` is most often used to merely do delta expansion (though I've never actually heard anyone call it delta expansion), and it may even be true that the intent of `fold` is to do delta expansion and everything else is sort-of an accidental side effect of how fold is implemented, but delta reduction is not enough to undo the effects of fold in general (consider `fold (1 + 1)` in a goal containing `2`), and thus it cannot be said that `fold` only does delta-expansion.  Furthermore, if you have multiple layers of definitions and have only unfolded the top-most layer (e.g., define a to be b and define b to be c, and then you've unfolded a), fold will not be able to recognize that and undo the delta reduction because it uses `red`.

> Is this a bit like fold in that the set of one_terms are first beta-reduced?

I don't think they're beta-reduced, but I would have to check.  In any case, the terms being given to pattern are terms to look for in the goal (which will also appear in the result), not terms to manipulate to become patterns.

Reply:
Sorry about that.

Reply:
No worries.